portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java

```diff
@@ -872,13 +872,18 @@ public Group deleteGroup(Group group) throws PortalException {
 				userGroupGroupRoleLocalService.
 					deleteUserGroupGroupRolesByGroupId(group.getGroupId());
 
-				if (!group.isStagingGroup() &&
-					(group.isOrganization() || group.isRegularSite())) {
```

Comment:
This clause was added in March by @arbetts in b70e88734ce4e3f.

Andrew, are you ok with removing the clause and deleting the resource always? What was the reasoning for doing it only for some types of groups?